src/town_cmd.cpp

```diff
@@ -2879,7 +2880,6 @@ static bool TryBuildTownHouse(Town *t, TileIndex tile)
 
 CommandCost CmdPlaceHouse(DoCommandFlag flags, TileIndex tile, HouseID house)
 {
-	if (_game_mode != GM_EDITOR) return CMD_ERROR;
```

Comment:
With commands you have to assume the worst since they are coming from the network and you can't trust that. Yes, the other client might have said it's a release version, but it might be modified to change the UI check so 'Place house' is always shown.

Without guarding in this command, that would mean that player is "allowed" to place a house, while those playing fairly are not allowed to place houses.